linkis-web-next/src/pages/parameterConfig/drawer.vue

```diff
@@ -16,491 +16,531 @@
+// Make sure the version is matched to the engineType
+const handleChangeEngineType = (type: any) => {
+    tmpData.value.childCategory.forEach((cate: any) => {
+        if (cate.engineType === type) {
+            cate.version = engineTypeVersions.value[type][0].value;
+        }
+    })
+}
+
 const open = () => {
     show.value = true;
 };
 
+const deleteEngine = (index: number) => {
+    tmpData.value.childCategory.splice(index, 1);
+}
+
 const addEngine = () => {
-    engineConfigList.push({
-        type: '',
+    tmpData.value.childCategory.push({
+        engineType: '',
         version: '',
         description: '',
-    });
+    })
 };
 
 const confirmEditing = () => {
-    // TODO: send a request to save the config
-    show.value = false;
+    tmpData.value.categoryName = trim(tmpData.value.categoryName);
+    let isBlank = false;
+    tmpData.value.childCategory.forEach((cate: any) => {
+        if(cate.engineType === '' || cate.version === '' || !cate.version) {
+            FMessage.error(t('message.linkis.tableColumns.engineVersionCannotBeNull'));
+            isBlank = true;
+        }
+    })
+    if (isBlank) return;
+    const distinctEngineTypeList = [...new Set(tmpData.value.childCategory.map((cate: any) => cate.engineType))];
+    if (distinctEngineTypeList.length !== tmpData.value.childCategory.length) {
+        FMessage.error(t('message.linkis.nonRepetitiveEngines'))
+        return;
+    }
+    if (isAddingApplicationAndEngine.value) {
+        if (tmpData.value.categoryName.length === 0) {
+            FMessage.error(t('message.linkis.tableColumns.addApplicationRules'));
+            return;
+        }
+    }
+    FModal.confirm({
+        title: t('message.common.userMenu.title'),
+        content: t('message.linkis.confirmToModifyCreator'),
+        closable: true,
+        onOk: async () => {
+            if (isAddingApplicationAndEngine.value) {
+                try {
+                    await api.fetch("/configuration/createFirstCategory", {
+                        categoryName: tmpData.value.categoryName,
+                        description: tmpData.value.description,
+                    }, "post")
+                    FMessage.success(t('message.linkis.successfullyAddCreator'));
+                } catch (error) {
+                    FMessage.error(t('message.linkis.failedToAddCreator'));
+                    return;
+                }
+                let categoryId;
+                try {
+                    const res = await api.fetch("/configuration/getCategory", "get");
+                    categoryId = res.Category.find((v: any) => v.categoryName === tmpData.value.categoryName).categoryId;
+                } catch (error) {
+                    FMessage.error(t('message.linkis.failedToGetCategoryId'))
+                    await refresh();
+                    return;
+                }
+                for (const cate of tmpData.value.childCategory) {
+                    try {
```

Comment:
If there are no chain dependencies in the requests here, it might be more efficient to use Promise.all to execute them in parallel.

Reply:
It would be better if similar logic could be wrapped into a function.